docs/structure-of-a-contract.rst

```diff
@@ -42,6 +42,41 @@ Functions are the executable units of code within a contract.
 and have different levels of visibility (:ref:`visibility-and-getters`)
 towards other contracts. Functions must be decorated with either @public or @private.
 
+Default function
+----------------
+
+A contract can also have a default function, which is executed on a call to the contract if no other functions match the given function identifier (or if none was supplied at all, such as through someone sending it Eth). It is the same construct as fallback functions `in Solidity <https://solidity.readthedocs.io/en/latest/contracts.html?highlight=fallback#fallback-function>`_. 
+
+This function is always named `__default__` and must be annotated with `@public` and `@payable`. It cannot have arguments and cannot return anything.
```

Comment:
I think `@payable` is optional, confirm @jacqueswww?

Reply:
Nope, documented in the unit test case.

Reply:
I'm not :100:, wait for confirmation from @jacqueswww unless you found in the code/tests where this is allowed.

Reply:
@fubuloubu Haha I found the test, I just misread the failure condition initially.

https://github.com/ethereum/vyper/blob/master/tests/parser/functions/test_default_function.py#L34

Reply:
Yes payable is optional, depending on the use case ;)